### PR TITLE
[licensepp] Add new port

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- What does your PR fix? Add new port https://github.com/zuhd-org/licensepp.

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@myd7349, thanks for the PR!

The following triplets failed on CI testing, is this expect?
licensepp:arm-uwp
licensepp:x64-uwp

**Comment by JackBoosY** [comment]
Will restart CI test after #11742 merged.

**Review by PhoebeHui — APPROVED** [review]

File: ports/licensepp/fix-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
++if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
++    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O3 -Wall -Werror")
+ endif()
+ 
+ # Check for cryptopp (static)
+@@ -64,15 +62,12 @@ set(LICENSEPP_SOURCE_FILES
+     src/license.cc
+ )
+ 
+-if (BUILD_SHARED_LIBS)
```
Comment (JackBoosY) [inline ports/licensepp/fix-cmake.patch:22]: We can revert this change because it is equivalent to before the modification.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!
